src/main/MainScreenContainer.js

```diff
@@ -40,11 +40,11 @@ class MainScreenContainer extends React.Component {
     }
   }
 
-  doNarrow = (newNarrow = [], anchor: number = Number.MAX_SAFE_INTEGER) => {
+  doNarrow = (newNarrow = [], anchor: number = Number.MAX_SAFE_INTEGER, message) => {
     const { auth, switchNarrow, messages } = this.props;
     registerAppActivity(auth);
     requestIdleCallback(() =>
-      switchNarrow(newNarrow, messages.filter(msg => msg.id === anchor)
+      switchNarrow(newNarrow, message || messages.filter(msg => msg.id === anchor)
```

Comment:
right, `message` is just an array
I didn't get the point here. Since i have used getMessages, i think the newly fetched message won't be in the `messages` array.
We could and may be should update the message state after getMessages to reflect this new message before calling doNarrow?